src/dotnet-core-uninstall/Shared/VSVersioning/VisualStudioSafeVersionsExtractor.cs

```diff
@@ -48,5 +53,28 @@ private static IEnumerable<IEnumerable<Bundle>> DivideSpecialCases(IEnumerable<B
                 bundleList.GroupBy(bundle => bundle.Version.SemVer.Patch < division.Patch) as IEnumerable<IEnumerable<Bundle>> :
                 new List<IEnumerable<Bundle>> { bundleList };
         }
+
+        public static Dictionary<Bundle, string> GetListCommandUninstallableStrings(IEnumerable<Bundle> allBundles)
+        {
+            var uninstallable = GetUninstallableBundles(allBundles);
+            var required = allBundles.Where(b => !uninstallable.Contains(b));
+
+            var ListCommandStringResults = uninstallable.Select(b => new KeyValuePair<Bundle, string>(b, string.Empty))
+                        .ToDictionary(i => i.Key, i => i.Value);
+            if (required.Where(b => b.Version.SemVer < SpecialCaseDivisions[0]).Count() > 0) 
+            {
+                ListCommandStringResults.Add(required.Where(b => b.Version.SemVer < SpecialCaseDivisions[0]).Max(), "[Required by Visual Studio 2017]");
+            }
+            foreach (var recentSdk in required.Where(b => b.Version.SemVer >= UpperLimit))
+            {
+                ListCommandStringResults.Add(recentSdk, $"[Cannot uninstall version {UpperLimit} and above]");
```

Comment:
I'm going to talk more to Kathleen about this when she is back tomorrow, since I agree that it would be helpful to have more information about why SDKs aren't allowed to be uninstalled but we don't want these strings to create clutter in the list command output. For now I've added more information when verbosity is set to anything above normal.